src/core/DataFile.cpp

```diff
@@ -1790,6 +1790,30 @@ void DataFile::upgrade_mixerRename()
 }
 
 
+// Rename BB to pattern and TCO to clip
+void DataFile::upgrade_bbTcoRename()
+{
+	std::vector<std::pair<const char *, const char *>> names {
+		{"tco", "clip"},
```

Comment:
No corresponding `nodeName()`? Also, I can't find any XML tag "tco" in any of my LMMS projects from the last 10 years :smiley:

Reply:
:upside_down_face: this was interesting, I've confused it with the `"clips"` used by drag-n-drop, but that needs no upgrading!